### ISSUE TITLE
Interface method wrongly being flagged as PHP4 legacy constructor

### ISSUE THREAD

**Issue by darkain** [facebook/hhvm#8121]
### HHVM Version
HipHop VM 3.25.0-dev (rel) - Travis CI "hhvm-nightly"

### Operating System and Version
Travis CI
Runtime kernel version: 4.14.12-041412-generic
travis-build version: 8d2b1ed85
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.5 LTS
Release:	14.04
Codename:	trusty

### Standalone code, or other way to reproduce the problem
```
interface test {
	public function test();
}
```

### Expected result
Normal compilation and execution without issue. This is an interface, not a class, so there shouldn't be an assumed constructor from using the same method name.

### Actual result
Fatal error: Uncaught exception 'ErrorException' with message 'Methods with the same name as their class will not be constructors in a future version of PHP; test has a deprecated constructor' in /path/to/file.php:18

**Comment by fredemmott** [comment]
With both your example and:

```
<?php

interface test {
	public function test();
}

class foo implements test {
  public function test() {
    var_dump('hello');
  }
}

(new foo())->test();
```

.. on both 3.25.0 and 3.25.1 (using Facebook's public docker images):

```
root@9c0749062f8d:/# hhvm /tmp/test.php
string(5) "hello"
root@9c0749062f8d:/# hhvm -d hhvm.php7.all=0 /tmp/test.php
string(5) "hello"
root@9c0749062f8d:/# hhvm -d hhvm.jit=0 /tmp/test.php
string(5) "hello"
root@9c0749062f8d:/# hhvm -d hhvm.jit=1 /tmp/test.php
string(5) "hello"
```

**Comment by fredemmott** [comment]
also not reproducible on Travis: https://travis-ci.org/fredemmott/travistest/builds/357455454

Can you share a reproducible example of the problem?

**Comment by darkain** [comment]
https://travis-ci.org/darkain/pudl/jobs/353926098

This is where it is failing. But even having just a PHP file with the code listed above (just the interface, not even using it) is causing the error for me. The interface cannot even be parsed properly without generating this blocking error.

**Comment by fredemmott** [comment]
@darkain: can you:
 - open a new terminal
 - do `cat testfile.php` where this is just the code listed above
 - run it with HHVM
 - try also running it with `hhvm --no-config`
 - copy-paste the entire terminal session in to here please?

**Comment by fredemmott** [comment]
If I grab the first 26 lines of https://github.com/darkain/pudl/blob/master/pudlInterfaces.php, HHVM seems happy with that.

**Comment by darkain** [comment]
It has to deal with error reporting apparently. Due to the strict nature of the environment my code runs in, we do not allow warnings at any level. Setting error reporting to E_ALL will generate the initial warning, then we upgrade warnings to fatal exceptions. Either way, regardless of severity, HHVM is improperly throwing a code deprecation warning in an area where it shouldn't be.

https://travis-ci.org/darkain/travistest/jobs/357519505

**Comment by fredemmott** [comment]
# Why I wasn't seeing it:

I  can reproduce with either --no-config or `-d hhvm.log.runtime_error_reporting_level=8192`

We've been setting it to 8191 since 2013, when these errors didn't exist. I'm not sure why.

# What will be changed

- Yes, this should *only* be raised for classes, not traits or interfaces.
- I'll remove the 8191 setting from the nightly builds and see what reports we get. Not including PHP7 deprecation notices by default (even ones that are incorrect, like this),is unintentional

# What had changed

It doesn't look like anything changed in 3.25 around this specific error - however, PHP7 mode was enabled by default, and this is (meant to be) a PHP7 deprecation notice.

# Workarounds

- set hhvm.php7.all=0 in your INI file
- HHVM doesn't raise this as a fatal error; you're doing that here, and could handle it: https://github.com/darkain/pudl/blob/eabdc055eae9870754c480ea31ed942d98fafb92/test/cli.php#L6

# Supporting HHVM in this project

Not specific for this issue, but you probably want to read https://hhvm.com/blog/2017/09/18/the-future-of-hhvm.html

**Comment by fredemmott** [comment]
Internal fix ID: D7418115
